concepts section with docs pages for Endpoint, Relay, Protocol, Router (#241)
n0-computer/iroh.computer: src/app/docs/concepts/endpoint/page.mdx

Somewhere in a program that uses iroh for direct connections, you'll see code like this:

```rust
use iroh::Endpoint;

#[tokio::main]
async fn main() {
    let endpoint = Endpoint::builder().bind().await.unwrap();
    // ...
}
```

Breaking that down the `builder` sets up configuration for the endpoint, and `bind` creates the endpoint and starts listening for incoming connections. The `await` keyword is used to wait for the endpoint to be created in an asyncronous context.

Once you have an endpoint, you can use it to create connections to other nodes, or accept incoming connections from other nodes.

## Connections

Because we're in a peer-2-peer context, either node might be operating as the "server", so we use `connect` and `accept` to distinguish between the two. The `connect` method is used to create a new connection to a remote node, while `accept` is used to accept incoming connections from a remote node.

Connections are full-fledged QUIC connections, giving you access to most features of QUIC / HTTP3, including bidirectional and unidirectional streams.

<Note>

> QUIC streams are due to the light-weight properties of streams a stream will only be accepted once the initiating peer has sent some data on it.

Reviewer: ```suggestion
Due to the light-weight properties of QUIC streams a stream can only be accepted once the initiating peer has sent some data on it.
```
